### PR TITLE
Paddle Video V2

### LINKED ISSUE #2008: 视频模型评估预测问题
1.stnet,tsn 运行test.py时会报显存不足，建议调低configs文件中TEST默认batch_size大小
2.视频模型库中好几个模型readme infer.py中示例命令行的模型名都是小写，运行后会报错：
64f7f729746da3ad61898579d45f308a

### PR DESCRIPTION
**Pull request by SunGaofeng**
V2 of Paddle Video
新增模型：Non-local, TSM

fix https://github.com/PaddlePaddle/models/issues/2008

### CODE REVIEW

File: PaddleCV/video/models/nonlocal_model/README.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+### 具体实现
+
+下图描述了Non-local block使用内嵌高斯形式关联函数的具体实现过程，
+<p align="center">
+<img src="../../images/nonlocal_instantiation.png" height=488 width=585 hspace='10'/> <br />
+使用Eembedded Gaussian关联函数的Non-local block
+</p>
+
+g(Xj)是对输入feature map做一个线性变换，使用1x1x1的卷积；theta和phi也是线性变化，同样使用1x1x1的卷积来实现。从上图中可以看到，Non-local操作只需用到通常的卷积、矩阵相乘、加法、softmax等比较常用的算子，不需要额外添加新的算子，用户可以非常方便的实现组网以构建模型。
```
Comment (heavengate) [inline PaddleCV/video/models/nonlocal_model/README.md:77]: theta -> $\theta$ ?

File: PaddleCV/video/models/nonlocal_model/README.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+#### Concatenation
+
+<p align="center">
+<a href="https://www.codecogs.com/eqnedit.php?latex=f(x_i,&space;x_j)&space;=&space;ReLU(w_f^T[\theta(x_i),\phi(x_j)]),&space;\qquad&space;C(x)&space;=\mathit{N}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?f(x_i,&space;x_j)&space;=&space;ReLU(w_f^T[\theta(x_i),\phi(x_j)]),&space;\qquad&space;C(x)&space;=\mathit{N}" title="f(x_i, x_j) = ReLU(w_f^T[\theta(x_i),\phi(x_j)]), \qquad C(x) =\mathit{N}" /></a>
+</p>
+
+其中
+<p align="center">
+<a href="https://www.codecogs.com/eqnedit.php?latex=\theta(x_i)=W_{\theta}x_i,&space;\qquad&space;\phi(x_j)=W_{\phi}x_j" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\theta(x_i)=W_{\theta}x_i,&space;\qquad&space;\phi(x_j)=W_{\phi}x_j" title="\theta(x_i)=W_{\theta}x_i, \qquad \phi(x_j)=W_{\phi}x_j" /></a>
+</p>
```
Comment (heavengate) [inline PaddleCV/video/models/nonlocal_model/README.md]: can here use $..math..$ ?

File: PaddleCV/video/models/nonlocal_model/README.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+<a href="https://www.codecogs.com/eqnedit.php?latex=y_{i}=\frac{1}{C(x)}&space;\sum_{j}f(x_i,&space;x_j)g(x_j)" target="_blank"><img src="https://latex.codecogs.com/gif.latex?y_{i}=\frac{1}{C(x)}&space;\sum_{j}f(x_i,&space;x_j)g(x_j)" title="y_{i}=\frac{1}{C(x)} \sum_{j}f(x_i, x_j)g(x_j)" /></a>
+</p>
+
+在上面的公式中，x表示输入feature map， y表示输出feature map，i是输出feature map的位置，j是输入feature map的位置，f(xi, xj)描述了输出点i跟所有输入点j之间的关联，C是根据f(xi, xj)选取的归一化函数。g(xj)是对输入feature map做一个变换操作，通常可以选取比较简单的线性变换形式；f(xi, xj)可以选取不同的形式，通常可以使用如下几种形式
+
+#### Gaussian
+
+<p align="center">
+<a href="https://www.codecogs.com/eqnedit.php?latex=f(x_i,&space;x_j)&space;=&space;e^{x_i^Tx_j},&space;\qquad&space;C(x)&space;=&space;\sum_{j}f(x_i,&space;x_j)" target="_blank"><img src="https://latex.codecogs.com/gif.latex?f(x_i,&space;x_j)&space;=&space;e^{x_i^Tx_j},&space;\qquad&space;C(x)&space;=&space;\sum_{j}f(x_i,&space;x_j)" title="f(x_i, x_j) = e^{x_i^Tx_j}, \qquad C(x) = \sum_{j}f(x_i, x_j)" /></a>
+</p>
```
Comment (heavengate) [inline PaddleCV/video/models/nonlocal_model/README.md:34]: can here use $..math..$

File: PaddleCV/video/README.md [hunk 4]
```diff
@@ -10,17 +10,19 @@
...
 | [NeXtVLAD](./models/nextvlad/README.md)  | 视频分类| 2nd-Youtube-8M最优单模型 |
 | [StNet](./models/stnet/README.md)  | 视频分类| AAAI'19提出的视频联合时空建模方法 |
+| [TSM](./models/tsm/README.md) | 视频分类| 基于时序移位的简单高效视频时空建模方法 |
 | [TSN](./models/tsn/README.md) | 视频分类| ECCV'16提出的基于2D-CNN经典解决方案 |
+| [Non-local](./models/nonlocal_model/README.md) | 视频分类| 视频非局部关联建模模型 |
 
 ### 主要特点
 
-- 包含视频分类方向的多个主流领先模型，其中Attention LSTM，Attention Cluster和NeXtVLAD是比较流行的特征序列模型，TSN和StNet是两个End-to-End的视频分类模型。Attention LSTM模型速度快精度高，NeXtVLAD是2nd-Youtube-8M比赛中最好的单模型, TSN是基于2D-CNN的经典解决方案。Attention Cluster和StNet是百度自研模型，分别发表于CVPR2018和AAAI2019，是Kinetics600比赛第一名中使用到的模型。
+- 包含视频分类方向的多个主流领先模型，其中Attention LSTM，Attention Cluster和NeXtVLAD是比较流行的特征序列模型，Non-local, TSN, TSM和StNet是End-to-End的视频分类模型。Attention LSTM模型速度快精度高，NeXtVLAD是2nd-Youtube-8M比赛中最好的单模型, TSN是基于2D-CNN的经典解决方案。Attention Cluster和StNet是百度自研模型，分别发表于CVPR2018和AAAI2019，是Kinetics600比赛第一名中使用到的模型。
```
Comment (SunGaofeng) [inline PaddleCV/video/README.md]: 添加non-local和tsm的描述

File: PaddleCV/video/test.py [hunk 5]
```diff
@@ -44,19 +44,19 @@ def parse_args():
         default='configs/attention_cluster.txt',
         help='path to config file of model')
     parser.add_argument(
-        '--batch-size',
+        '--batch_size',
         type=int,
         default=None,
         help='traing batch size per GPU. None to use config file setting.')
```
Comment (SunGaofeng) [inline PaddleCV/video/test.py]: traing -> test

File: PaddleCV/video/models/nonlocal_model/README.md (hunk 3 above)
Comment (SunGaofeng) [reply PaddleCV/video/models/nonlocal_model/README.md:34]: 试了下这么做显示不出公式

File: PaddleCV/video/models/nonlocal_model/nonlocal_model.py [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+            logger.info(name)
+    return common_names
+
+
+def load_params_from_file(exe, prog, pretrained_file, place):
+    logger.info('load params from {}'.format(pretrained_file))
+    if '.pkl' in pretrained_file:
+        load_params_from_pkl_file(prog, pretrained_file, place)
+    else:
+        load_params_from_paddle_file(exe, prog, pretrained_file, place)
```
Comment (qingqing01) [inline PaddleCV/video/models/nonlocal_model/nonlocal_model.py]: 上面复杂的params load需要修改~

Comment (SunGaofeng) [reply PaddleCV/video/models/nonlocal_model/nonlocal_model.py]: yes, removed load_params_from_pkl_file, and move load_params_from_file to another file nonlocal_utils.py

**Review by heavengate — APPROVED** [review]
LGTM

**Pull request merged**